placement/algo/sharded.go

```diff
@@ -168,11 +168,7 @@ func (a rackAwarePlacementAlgorithm) ReplaceInstances(
 			return nil, err
 		}
 		load := loadOnInstance(leavingInstance)
-		if load == 0 {
-			result, _, err := addInstanceToPlacement(ph.GeneratePlacement(), leavingInstance, withShards)
-			return result, err
-		}
-		if !a.opts.AllowPartialReplace() {
+		if load != 0 && !a.opts.AllowPartialReplace() {
```

Comment:
the changes in this file, glide, and `kv/util/...` seem unrelated to TLS changes. Any reason to group them in this PR?

Reply:
rebased from master to try and figure out why CI was barfing, i'll undo the rebase